library/src/main/java/com/getstream/sdk/chat/view/messageinput/AttachmentsController.kt

```diff
@@ -0,0 +1,360 @@
+package com.getstream.sdk.chat.view.messageinput
+
+import android.net.Uri
+import androidx.annotation.VisibleForTesting
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.adapter.FileAttachmentListAdapter
+import com.getstream.sdk.chat.adapter.FileAttachmentSelectedAdapter
+import com.getstream.sdk.chat.adapter.MediaAttachmentAdapter
+import com.getstream.sdk.chat.adapter.MediaAttachmentSelectedAdapter
+import com.getstream.sdk.chat.enums.MessageInputType
+import com.getstream.sdk.chat.infrastructure.DispatchersProvider
+import com.getstream.sdk.chat.model.AttachmentMetaData
+import com.getstream.sdk.chat.model.ModelType
+import com.getstream.sdk.chat.utils.Constant
+import com.getstream.sdk.chat.utils.PermissionHelper
+import com.getstream.sdk.chat.utils.StorageHelper
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+internal class AttachmentsController(
+    private val rootController: MessageInputController,
+    private val permissionHelper: PermissionHelper,
+    private val storageHelper: StorageHelper,
+    private val dispatchersProvider: DispatchersProvider,
+    private val view: MessageInputView,
+    private val showOpenAttachmentsMenuConfig: Boolean
+) {
+    @VisibleForTesting
```

Comment:
It will help use to update the internal implementation without touch our tests and be able to know that our new implementation works like the previous one

Reply:
Rewrote it. Now adapters are provided via constructors. It makes logic more direct and clean